src/electronic-ids/pcsc/pcsc-common.hpp

```diff
@@ -47,24 +47,23 @@ inline pcsc_cpp::byte_vector addPaddingToPin(pcsc_cpp::byte_vector&& pin, size_t
                                              pcsc_cpp::byte_type paddingChar)
 {
     pin.resize(std::max(pin.size(), paddingLength), paddingChar);
-    return pin;
+    return std::move(pin);
```

Comment:
```suggestion
    return pin;
```
This is not needed

Reply:
I disagree, see https://stackoverflow.com/a/14857144. When you have a function that takes an rvalue reference, modifies it, and then returns a copy of it, `std::move()` should be used as far as I understand. Yakk-Adam Nevraumont mentions that in C++20 the behavior changes, but I would like to see a specific reference and explanation of the new behavior.

`pin` is an lvalue within the function scope, even though it's an rvalue reference parameter. Returning an lvalue would typically invoke a copy - and yes, this _may_ be elided with NRVO as you imply, but by explicitly applying `std::move`  we can be sure that it is moved as intended and not rely on probable behavior.

Reply:
By the way, we should deal with the `resize()` problem here as well.